ignite/engine/__init__.py

```diff
@@ -91,20 +101,54 @@ def create_supervised_trainer(
 
     device_type = device.type if isinstance(device, torch.device) else device
     on_tpu = "xla" in device_type if device_type is not None else False
+    has_native_amp, has_apex_amp = False, False
 
     if on_tpu and not idist.has_xla_support:
         raise RuntimeError("In order to run on TPU, please install PyTorch XLA")
 
+    if amp and LooseVersion(torch.__version__) >= LooseVersion("1.6.0"):
+        from torch.cuda.amp import autocast
+
+        has_native_amp = True
+    elif amp:
+        try:
+            from apex import amp as apex_amp
+
+            has_apex_amp = True
```

Comment:
maybe we can even initialize model and optimizer here with amp ?

Reply:
As per apex docs, we can't do that.
> amp.initialize should be called **after** you have finished constructing your model(s) and optimizer(s), but **before** you send your model through any DistributedDataParallel wrapper. See Distributed training in the Imagenet example.

`amp.initialize` should be called **after** model construction, but **before** setting DDP or DP, so we have no way to do that if user pass output of `idist.auto_model(model)`.

And I think, given user to initialize model is the cleanest way in terms of code and usability.

Reply:
Yes, make sense. Thanks!